pwa-shell/src/ui/login/LoginFormContainer.js

```diff
@@ -2,8 +2,14 @@ import { connect } from 'react-redux';
 import { login as performLogin } from 'state/thunks';
 import LoginForm from 'ui/login/LoginForm';
 
-const mapDispatchToProps = dispatch => ({
-  onSubmit: data => dispatch(performLogin(data)),
+const mapDispatchToProps = (dispatch, { history, location }) => ({
+  onSubmit: data =>
+    dispatch(performLogin(data)).then(() => {
+      const redirectURI = new URLSearchParams(location.search).get(
+        'redirect_uri',
+      );
+      history.replace(redirectURI ? redirectURI : '/');
```

Comment:
why `replace` instead of `push`?

Reply:
my idea was after logging in, then it proceeds to the content detail and should the user chooses to go back to the listing (usually having a habit by pressing back), they won't be going back to login screen